### PR TITLE
Update GitHub Actions

### PR DESCRIPTION
**Pull request by nickelc**
The PR fixes all deprecation warnings from the build job.

The toolchain is now installed with [`dtolnay/rust-toolchain`](https://github.com/dtolnay/rust-toolchain) and the binaries are installed with [`taiki-e/install-action`](https://github.com/taiki-e/install-action).

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (djc) [inline .github/workflows/ci.yml:42]: For deny, I think it would make more sense to use Embark's cargo-deny action directly?

Comment (nickelc) [reply .github/workflows/ci.yml:42]: I don't know. It does the same but with the docker overhead.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
This looks good to me. @ctz do you have any security concerns about this?

**Comment by djc** [comment]
@cpu got any thoughts? @ctz ping?

File: .github/workflows/ci.yml
Comment (djc) [reply .github/workflows/ci.yml:42]: > The maintainers of install-action also appear a little bit more responsive than the cargo-deny folks.

In my experience the cargo-deny folks have usually been pretty responsive? (I'm okay with using this instead, though.)

Comment (cpu) [reply .github/workflows/ci.yml:42]: I have no actual experience with either upstream so defer to you :-) My comment is based on a very unscientific look at the open issues/PRs across the two repos and shouldn't be weighed very heavily.

**Review by ctz — APPROVED** [review]
For what it's worth, this repo (as well as all others in the rustls org) only give read permissions to their GH actions. That means any third-party actions we use only get the same access as everyone else on the internet. So (assuming github didn't screw up) this isn't really a security question but one of resilience/correctness/maintainability.

**Review by djc — APPROVED** [review]

**Pull request merged**
